### DISCUSSION TITLE
Multi-release jar?

### DISCUSSION THREAD

**Discussion by pbludov** [CaciocavalloSilano/caciocavallo#10] [Ideas]
Have you thought about releasing a universal version for all JDK using [multi-release jars](https://www.baeldung.com/java-multi-release-jar)?

**Comment by gschrader** [comment]
It was talked about in #1 a bit, back then I think the tooling wasn't quite there yet. It's maybe worth looking at it again.

**Comment by pbludov** [comment]
I'll try to explain what the issue is.

Our project is limited to Java 11 version, so we use `caciocavallo` version 1.11.
The dependabot immediately comes and offers a [upgrade to the version 1.17](https://github.com/checkstyle/checkstyle/pull/11500).
But this version will not work for our project. We have to turn off updates for `caciocavallo`.
And that's the problem. What if you want to release version 1.11.1 with some security fixes?
We will skip this version because we are used to relying on dependabot updates.

If you go the MRJAR way, the version will constantly increase, but at the same time compatibility with all supported versions of the JDK will be maintained.

**Reply by gschrader** [reply]
Yup I can see that being annoying. I'm not sure when I'll get a chance to have a look though. I assume there must be a way to tell depndabot to ignore certain libs which might be an option for the short term.

**Comment by gschrader** [comment]
I took a stab at a MRJAR, I'm still not sure the tooling is there. The existing project structure makes it a little tough as well so I have to push this off some more.
